### PR TITLE
Organize debugging documentation

### LINKED ISSUE #55660: docs: Provide detail about debugging qemu
**Describe the bug**
[The application development guide](https://docs.zephyrproject.org/latest/develop/application/index.html#application-debugging) describes how to start a debugging session for an application running on a QEMU host. It would be easier to follow the instructions if it provided additional details about running the debug server and accessing the console output.

In particular:
* `ninja debugserver` won't work unless there is a `qemu-fifo` file in the build directory, and you have to create one. (If I wasn't supposed to have to do that, that might be my error or a separate issue.)
* Having attached to the debug server, you can set breakpoints in the test and do various other things. Neither `gdb` nor the server will show the console output, so if you just `continue` after attaching, it will appear as though nothing happened.
* You can view the console output during the test with something like `tail -f qemu-fifo`.

**Impact**
I had to spend a couple hours messing around to debug a test on qemu usefully.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.15.2
 - Commit SHA or Version used: bc5fbc929bb805e3c562c7280dd02a6f9f1f9087

### PR DESCRIPTION
**Pull request by alevkoy**
Add some detail about debugging with QEMU. Streamline/clean up the text slightly.

Also create `qemu-fifo` during the build, so the documented `ninja debugserver` command works. (I realize this isn't really a documentation issue. I can make another issue for this if somebody wants.)

Fixes #55660

### CODE REVIEW

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please fix the CMake part.

**Comment by alevkoy** [comment]
OK, the "commit suggestion" button did not do what I was hoping it would. I'll use those as fixups and push again.

**Comment by alevkoy** [comment]
Suggestions fixed up and pushed.

**Review by tejlmand — DISMISSED** [review]
lgtm

File: doc/develop/application/index.rst [hunk 1]
```diff
@@ -1054,14 +1054,22 @@ environment.
 The simplest way to debug an application running in QEMU is using the GNU
 Debugger and setting a local GDB server in your development system through QEMU.
 
-You will need an Executable and Linkable Format (ELF) binary image for
+You will need an :abbr:`ELF (Executable and Linkable Format)` binary image for
 debugging purposes.  The build system generates the image in the build
-directory.  By default, the kernel binary name is
-:file:`zephyr.elf`. The name can be changed using a Kconfig option.
+directory.  By default, the kernel binary name is :file:`zephyr.elf`. The name
+can be changed using a Kconfig option.
```
Comment (nordicjm) [inline doc/develop/application/index.rst:1060]:
```suggestion
can be changed using :kconfig:option:`CONFIG_KERNEL_BIN_NAME`.
```

Comment (alevkoy) [reply doc/develop/application/index.rst]: Done.

**Review by tejlmand — APPROVED** [review]

**Comment by alevkoy** [comment]
@nordicjm Do you have any further comment?

**Review by nordicjm — APPROVED** [review]

**Pull request merged**